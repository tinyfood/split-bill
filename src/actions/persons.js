import uuid from 'uuid'

export const addPerson = (amount, locked = false) => ({
  type: 'ADD_PERSON',
  payload: {
    id: uuid(),
    amount,
    locked
  }
})

export const removePerson = ({ id } = {}) => ({
  type: 'REMOVE_PERSON',
  id
})

export const editPerson = (id, updates) => ({
  type: 'EDIT_PERSON',
  id,
  updates
})
