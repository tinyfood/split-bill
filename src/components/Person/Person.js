import React from 'react'
import styled from 'styled-components'

const PersonStyled = styled.div`
  height: ${props => (props.height ? props.height + 'px' : '100px')};
  background: grey;
  border: 1px solid black;
`

const Person = ({ amount }) => {
  return <PersonStyled height={100 / amount}>{amount}</PersonStyled>
}

export default Person
