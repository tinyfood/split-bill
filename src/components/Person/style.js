import styled from 'styled-components'

export const PersonStyled = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  flex: 1 1 0;
  border-bottom: 2px solid rgba(209, 216, 224, 0.41);
`

export const PersonId = styled.div`
  margin-bottom: 6px;
  font-size: 10px;
  font-weight: 700;
  text-transform: uppercase;
  letter-spacing: -0.25px;
  color: #4e4e4e;
`

export const PersonAmount = styled.div`
  font-size: 28px;
  font-weight: 700;
  color: #4e4e4e;
`
