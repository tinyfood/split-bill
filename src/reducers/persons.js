const personsReducerDefaultState = [
  {
    id: 1,
    amount: 0,
    locked: false
  },
  {
    id: 2,
    amount: 0,
    locked: false
  },
  {
    id: 3,
    amount: 0,
    locked: false
  }
]

export default (state = personsReducerDefaultState, action) => {
  switch (action.type) {
    case 'ADD_PERSON':
      return [...state, action.payload]
    case 'REMOVE_PERSON':
      return state.filter(({ id }) => id !== action.id)
    case 'EDIT_PERSON':
      return state.map(person => {
        if (person.id === action.id) {
          return { ...person, ...action.updates }
        } else {
          return person
        }
      })
    default:
      return state
  }
}
