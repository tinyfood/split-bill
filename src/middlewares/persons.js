import { calculateTipsAmount, calculateTotal } from '../actions/split'

const personsMiddleware = ({ dispatch }) => next => action => {
  next(action)

  // Add action type check

  switch (action.type) {
    case 'EDIT_FRIENDS':
      alert('EDITED FRIENDS')
      break
    default:
      return
  }
}

export default personsMiddleware
