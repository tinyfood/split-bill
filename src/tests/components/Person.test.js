import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import Person from '../../components/Person/Person'
import persons from '../fixtures/persons'

test('should render Person correctly', () => {
  const wrapper = shallow(<Person {...persons[0]} />)
  expect(toJSON(wrapper)).toMatchSnapshot()
})
