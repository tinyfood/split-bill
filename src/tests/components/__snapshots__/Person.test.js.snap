// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Dots correctly 1`] = `
<styled.li>
  <styled.div>
    Person 
    a
  </styled.div>
  <styled.div>
    $
    0
  </styled.div>
</styled.li>
`;

exports[`should render Person correctly 1`] = `
<styled.li>
  <styled.div>
    Person 
    a
  </styled.div>
  <styled.div>
    $
    0
  </styled.div>
</styled.li>
`;
